src/lib/CMakeLists.txt

```diff
@@ -312,6 +312,8 @@ set(
     optimizer/strategy/predicate_reordering_rule.hpp
     optimizer/strategy/predicate_split_up_rule.cpp
     optimizer/strategy/predicate_split_up_rule.hpp
+    optimizer/strategy/semi_join_reduction_rule.cpp
```

Comment:
I think we should invest more time in understanding why TPC-H Q21 degrades that much.

Reply:
Already did. The problem is that we cannot estimate ColumnVsColumnTableScans, so either we get lucky and have a good result, or we don't.

Reply:
Ok ... than question for another PR/issue: is this something we could fix with the existing histograms etc.?

Reply:
Probably, but I don't expect it to be trivial